lib/frrscript.h

```diff
@@ -97,16 +122,17 @@ void frrscript_register_type_codecs(struct frrscript_codec *codecs);
  */
 void frrscript_init(const char *scriptdir);
 
-#define ENCODE_ARGS(name, value)                                               \
-	do {                                                                   \
-		ENCODE_ARGS_WITH_STATE(L, value);                              \
-		lua_setglobal(L, name);                                        \
-	} while (0)
+#define ENCODE_ARGS(name, value) ENCODE_ARGS_WITH_STATE(lfs->L, value)
 
 #define DECODE_ARGS(name, value)                                               \
 	do {                                                                   \
-		lua_getglobal(L, name);                                        \
-		DECODE_ARGS_WITH_STATE(L, value);                              \
+		lua_getfield(lfs->L, 1, name);                                 \
+		if (lua_isnil(lfs->L, 2)) {                                    \
```

Comment:
some comment regarding indices, and its especially applicable here given we don't know (ostensibly anyway) where this macro will be used

Reply:
updated in https://github.com/FRRouting/frr/pull/8982/commits/70d995abd4e6938dcdaccc61a6f9cd66946d99d1